app/parsers/stackdriver.py

```diff
@@ -288,17 +288,18 @@ def add_resource():
             if not resource_data['name'].startswith(compute_reserved_prefixes):
                 add_resource()
 
-            # Also add the disk as a resource since theres not a separate log message for these
-            disk_name = prop("protoPayload.request.disks[?boot].diskName | [0]")
+            # Also add disk resources since theres not a separate log message for these
+            disk_names = prop('protoPayload.request.disks[*].initializeParams.diskName') or []
 
-            resource_data = {
-                'resource_type': 'compute.googleapis.com/Disk',
-                'name': disk_name or prop("protoPayload.resourceName").split('/')[-1],
-                'location': prop("resource.labels.zone"),
-                'project_id': prop("resource.labels.project_id"),
-            }
+            for disk_name in disk_names:
+
+                resource_data = {
+                    'resource_type': 'compute.googleapis.com/Disk',
+                    'name': disk_name,
+                    'location': prop("resource.labels.zone"),
+                    'project_id': prop("resource.labels.project_id"),
+                }
 
-            if not resource_data['name'].startswith(compute_reserved_prefixes):
```

Comment:
why are you dropping this?

Reply:
I was thinking we bailed when we checked it on the instance above, but we don't. Lemme see if it makes sense to test it per-disk, or just bail if the instance has a hidden prefix

Reply:
Updated. Now if we see an instanced with a reserved (hidden) prefix, we return before looking for disks